openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
Can we rename this class?  To be picky, we would like all of our Transformers to be as "optimized" as possible.  For example:
* use the most appropriate graph traversal algorithm for their needs. 
* percolate information down the graph instead of dynamically traversing up.
* collect all performance-sensitive data in the collect phase
* etc etc